### PR TITLE
refactor: kubelet config file drop in to cse

### PR DESCRIPTION
**Pull request by alexeldeib**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/nodecustomdata.yml [hunk 1]
```diff
@@ -445,15 +445,6 @@ write_files:
   content: !!binary |
     {{GetVariableProperty "cloudInitData" "customSearchDomainsScript"}}
 
-{{- if IsKubeletConfigFileEnabled}}
```
Comment (alexeldeib) [inline parts/linux/cloud-init/nodecustomdata.yml:448]: when there are no more top level if statements in cloud init, I can stop...

Comment (alexeldeib) [reply parts/linux/cloud-init/nodecustomdata.yml:448]: basically need to be able to have this file be totally empty or everything come from vhd

File: parts/linux/cloud-init/nodecustomdata.yml [hunk 2]
```diff
@@ -445,15 +445,6 @@ write_files:
...
-- path: /etc/systemd/system/kubelet.service.d/10-componentconfig.conf
-  permissions: "0600"
-  encoding: gzip
```
Comment (ganeshkumar5699) [inline parts/linux/cloud-init/nodecustomdata.yml:451]: Does `encoding: gzip` need to be present in cse_config too?

Comment (alexeldeib) [reply parts/linux/cloud-init/nodecustomdata.yml:451]: no, that only applies to these files: https://github.com/Azure/AgentBaker/blob/master/pkg/agent/variables.go#L22-L52

**Review by cameronmeissner — APPROVED** [review]

**Review by ganeshkumar5699 — APPROVED** [review]

**Comment by alexeldeib** [comment]
something fishy, containerd drop in no longer getting picked up 🤔

File: parts/linux/cloud-init/artifacts/cse_main.sh [hunk 3]
```diff
@@ -45,6 +45,9 @@ done
 sed -i "/#HELPERSEOF/d" "${CSE_HELPERS_FILEPATH}"
 source "${CSE_HELPERS_FILEPATH}"
 
+# for drop ins, so they don't all have to check/create the dir
+mkdir -p "/etc/systemd/system/kubelet.service.d"
```
[the remark this answers is not in the document]
Comment (alexeldeib) [reply parts/linux/cloud-init/artifacts/cse_main.sh]: somehow this doesn't work for containerd now, unclear

**Comment by alexeldeib** [comment]
fixed

**Pull request merged**